core/src/test/scala/unit/kafka/log/LogSegmentTest.scala

```diff
@@ -632,6 +633,15 @@ class LogSegmentTest {
     Utils.delete(tempDir)
   }
 
+  @Test
+  def testGetFirstBatchTimestampForSegments(): Unit = {
```

Comment:
It seems this test case does not cover `getFirstBatchTimestampForSegments`. Maybe it needs another name. Also, could you add UT for `getFirstBatchTimestampForSegments`?

Reply:
Thanks ! All comments are addressed in the latest commit.